Add docs links to /help and /info commands

saumya4751 opened the pull request:
This PR implements the changes requested in issue #63:

- Added DOCS_URL to config.py
- Created BotchCog abstract base class in interface/__init__.py
- Updated all cogs to inherit from BotchCog
- Set specific documentation URLs for cogs as specified in the issue
- Added documentation link buttons to the /help command
- Added a documentation link button to the /info command

All tests are passing with these changes.

saumya4751 committed:
Add docs links to /help and /info commands

saumya4751 committed:
Remove __pycache__ directories from tracking

tiltowait commented:
Hey, this looks pretty good! Some thoughts/ideas for improvements, now that I see the implementation:

1. Let's not import from the `src` package. You should be able to directly `import config`, for instance, due to how `pyproject.toml` is set up.
1. I know I said to make `BotchCog` an `ABC`, but after looking at it, let's make it into a `commands.Cog` subclass so all cogs inherit only from it (rather than multiple inheritance).
2. Once that's done, we can make an `@property` for `documentation_view` in `BotchCog`. If we have all cogs inherit off that, then we can lose the `hasattr` call and use `self.documentation_view` in both `/info` and the `/help` sections.
3. Feel free to add `__pycache__` to `.gitignore` if you like. I thought I had it in there, but apparently I just have it a global ignore file.

saumya4751 committed:
Implement feedback: Make BotchCog inherit from commands.Cog and add documentation_view property

saumya4751 commented:
Hi @tiltowait,

Thank you for the feedback! I've implemented all the suggested changes:

1. Changed imports to use direct module imports (e.g., `from config import DOCS_URL` instead of relative imports)
2. Made `BotchCog` inherit from `commands.Cog` so other cogs only need to inherit from `BotchCog`
3. Added a `documentation_view` property to `BotchCog` that returns a view with a documentation link button
4. Updated all cogs to use the new inheritance structure
5. Modified the help command and info command to use the `documentation_view` property
6. Added `__pycache__/` and related patterns to `.gitignore`

Let me know if there's anything else you'd like me to adjust!

tiltowait commented:
Looking much better! Just a couple of final things:

1. Please ensure [lint](https://github.com/tiltowait/botch/actions/runs/13931075747/job/39003486607?pr=78) passes (run `ruff check` if you have it; otherwise look at the link)
2. Please ensure [pyright passes](https://github.com/tiltowait/botch/actions/runs/13931075747/job/39003486610?pr=78) (I think on this one it would be fine to `cast(cog, BotchCog)` in `HelpCog.help_command()`)
3. The cogs in `src/interface/cofd/macros.py`, `src/interface/wod/basic.py`, and `src/interface/wod/macros.py` aren't inheriting from `BotchCog`.

These are pretty minor changes, so if you'd prefer, I can make them and merge. Just let me know. I'll give you credit in the changelog either way.

(Also, I don't care that the codecov/patch check fails. It's aspirational, and I really don't care about it for the `interface` dir.)

saumya4751 committed:
Fix lint and type errors, update all cogs to inherit from BotchCog

saumya4751 commented:
I've fixed all the issues mentioned:

- Resolved lint errors by removing unused imports
- Added type casting for BotchCog in the help command
- Updated all cogs to inherit from BotchCog

Verified everything with local ruff checks and testing. All should be good now!

tiltowait reviewed (approved):
Looks good! Thanks for tackling this!

tiltowait commented:
Merged in 5c829f6
